skcosmo/SparseMethods/Sparsified.py

```diff
@@ -0,0 +1,125 @@
+        :param coef0: free term of the polynomial and sigmoid kernel
+        :param kernel_params: kernel parameter set
+        :param n_active: the size of the small dataset used in learning
+        :param center: if True, centering of kernel during the learning is carried out
+        :param n_jobs: The number of jobs to use for the computation the kernel. This works by breaking down the pairwise matrix into n_jobs even slices and computing them in parallel.
+        """
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+        self.n_active = n_active
+        self.center = center
+        self.n_jobs = n_jobs
+        self._selector = selector
+
+    def _project(self, A, projector):
+        """Apply a projector to matrix A
+
+        :param A: projector: string corresponding to the named projection matrix of shape (a, p)
+        :type array-like, shape (n, a)
+
+        :return: A'
+        :type array-like, shape (n, p)
+        """
+        check_is_fitted(self, projector)
+
+        A = check_array(A)
+        A_transformed = np.dot(A, self.__dict__[projector])
+        return A_transformed
+
+    def _get_kernel(self, X, Y=None):
+        """
+        Calculate kernel for the matrix X or (optionally) for matrix X and Y
+        :param X: matrix, for which we calculate kernel
+        :param Y: A second feature array only if X has shape [n_samples_a, n_features].
+
+        :return: sklearn.metrics.pairwise.pairwise_kernels(X, Y)
+        """
+        if self.kernel == "precomputed":
+            if X.shape[-1] != self.n_active:
+                raise ValueError("The supplied kernel does not match n_active.")
+            return X
+        if callable(self.kernel):
+            params = self.kernel_params or {}
+        else:
+            params = {"gamma": self.gamma, "degree": self.degree, "coef0": self.coef0}
+        return pairwise_kernels(
+            X, Y, metric=self.kernel, filter_params=True, n_jobs=self.n_jobs, **params
+        )
+
+    def _define_kernel_matrix(self, X, X_sparse=None):
+        """
+        Calculate the Kmm and Knm matrices, which correspons to   kernel evaluated between the active set samples
+        and kernel matrix between X and X_sparse respectively
+        :param X: input matrices, for which we calculate Kmm and Knm
+        :param Kmm: kernel evaluated between the active set samples
+        :param Knm: kernel matrix between X and X_sparse
+        """
+        if X_sparse is None:
+            selector = self._selector(X)
+
+            i_active = selector.select(self.n_active)
+
+            X_sparse = X[i_active, :]
+
+            if self.kernel == "precomputed":
+                X_sparse = X_sparse[:, i_active]
+
+        self.X_sparse_ = X_sparse
+        K_sparse_ = self._get_kernel(self.X_sparse_, self.X_sparse_)
+
+        K_cross_ = self._get_kernel(X, self.X_sparse_)
```

Comment:
Why does this need a special name?

```suggestion
        K = self._get_kernel(X, self.X_sparse_)
```

Reply:
It just seems to me that we in the method as a whole are working with kernels. And denoting any of the kernels simply as `K` is not a good idea. It can lead to confusion.

Reply:
I think it's no more ridiculous to call the primary kernel `K` as it is to call `X` `X`

Reply:
I disagree. In sparse methods we have at least 3 different kernels  `K_MM`, `K_NM` and simple `K` - original kernel `K_NN`. So at the very least, there could already be confusion with `K_NN`

Reply:
Why are we not calling it K_NM then?

Reply:
Well, we named it in this way, but you said, what we need to rename it. Maybe it was because of `sklearn` compatibility, but I'm not sure. I'm totally agree, it's the best name

Reply:
So I did some diving -- `sklearn` either calls every "normal" kernel just `K` or uses a specific name (`K_gradient`, `K_new`, `K_diag`). I don't think `K_cross` gives the right idea for what we have here. If we want to be consistent with our literature, we need to call it `K_NM`. So I'll leave the decision to you -- would you rather call what is now `K_cross` `K_NM` or `K`? Either way, please go through and fix all instances.